tensorrt_llm/_torch/speculative/utils.py

```
@@ -549,11 +549,20 @@ def _build_spec_metadata(spec_config,
 
 def get_mtp_hidden_size(model_config) -> int:
     pretrained_config = getattr(model_config, "pretrained_config", model_config)
+    if getattr(pretrained_config, "model_type", None) == "qwen4_exp":
+        pretrained_config = pretrained_config.text_config
     hidden_size = getattr(pretrained_config, "hidden_size", None)
     if hidden_size is None:
         hidden_size = getattr(model_config, "hidden_size")
```

Ruff (0.16.3)

[warning] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)